src/lib.rs

```diff
@@ -1936,4 +2103,268 @@ mod test {
             &"sh(wsh(thresh(3,pk(Alice),s:pk(Bob),s:pk(Carol),sdv:older(2))))#l4qaawgv"
         );
     }
+
+    #[cfg(all(feature = "reserves", feature = "electrum"))]
+    #[test]
+    fn test_parse_produce_proof() {
+        let message = "Those coins belong to Satoshi Nakamoto";
+        let cli_args = vec![
+            "bdk-cli",
+            "--network",
+            "bitcoin",
+            "wallet",
+            "--descriptor",
+            "wpkh(cVpPVruEDdmutPzisEsYvtST1usBR3ntr8pXSyt6D2YYqXRyPcFW)",
+            "produce_proof",
+            "--message",
+            message.clone(),
+        ];
+
+        let cli_opts = CliOpts::from_iter(&cli_args);
+
+        let expected_cli_opts = CliOpts {
+            network: Network::Bitcoin,
+            subcommand: CliSubCommand::Wallet {
+                wallet_opts: WalletOpts {
+                    wallet: "main".to_string(),
+                    verbose: false,
+                    descriptor: "wpkh(cVpPVruEDdmutPzisEsYvtST1usBR3ntr8pXSyt6D2YYqXRyPcFW)"
+                        .to_string(),
+                    change_descriptor: None,
+                    #[cfg(feature = "electrum")]
+                    electrum_opts: ElectrumOpts {
+                        timeout: None,
+                        server: "ssl://electrum.blockstream.info:60002".to_string(),
+                        stop_gap: 10,
+                    },
+                    #[cfg(feature = "esplora")]
+                    esplora_opts: EsploraOpts {
+                        esplora: None,
+                        esplora_concurrency: 4,
+                    },
+                    #[cfg(any(feature = "compact_filters", feature = "electrum"))]
+                    proxy_opts: ProxyOpts {
+                        proxy: None,
+                        proxy_auth: None,
+                        retries: 5,
+                    },
+                },
+                subcommand: WalletSubCommand::OnlineWalletSubCommand(ProduceProof {
+                    msg: Some(message.to_string()),
+                }),
+            },
+        };
+
+        assert_eq!(expected_cli_opts, cli_opts);
+    }
+
+    #[cfg(all(feature = "reserves", feature = "electrum"))]
+    #[test]
+    fn test_parse_verify_proof() {
```

Comment:
I think a parsing test for "external descriptor" subcommand also needs to be added?